### DISCUSSION TITLE
Global list of Excluded IPs/Click Tracking Delay/Matomo integration

### DISCUSSION THREAD

**Discussion by tedo-develop** [shlinkio/discussions#42] [Ideas]
Hello,

first of all great work we like shlink! We have some suggestions for features.

**Global list of Excluded IPs** 
Would be great to have an interface in the settings tab where i can set ips to exclude from click-tracking.  

**click tracking delay**
Is there a way to enable/set a delay to count clicks/redirects?
At the moment every click is counted, even if i click multiple times in a short period.

**Matomo integration**
What do you think about a matomo integration? 
In some cases it is necessary to have statistics from a common accepted instance like matomo.

I know matomo allows to use campaigns for tracking, but this feature is not enough.
For example if you use campaigns for newsletters, only one klick will be counted and not every single klick in the newsletter.
Maybe it is a good idea to use events or content tracking for this use-case.

Thanks for your work!

**Comment by acelaya** [comment]
> first of all great work we like shlink!

Thank you!

> **Global list of Excluded IPs**
> Would be great to have an interface in the settings tab where i can set ips to exclude from click-tracking.

When you say the "settings tab", I guess you are talking about shlink-web-client. That tab is for the client itself, not for Slink.

However, Slink has it's own way to provide config options, so no problem.

Also, this shouldn't be super hard to do, so I'll try to get it added to some future version.

You may want to subscribe to https://github.com/shlinkio/shlink/issues/1104

> **click tracking delay**
> Is there a way to enable/set a delay to count clicks/redirects?
> At the moment every click is counted, even if i click multiple times in a short period.

This can be done already.

By default Shlink redirects visitors with a status 302, which tells the browser to hit the server every time they click the link.

However, this can be changed so that it uses a permanent redirect (301) with a certain time of cache.

With that, the browser will hit the server the first time, and just redirect on subsequent requests without hitting the server again.

The cache time is customizable, meaning, you can make only the first click to be tracked for 10min (for example), and the next click after that, coming from the same device, will be tracked again.

In order to change this, if you are using the docker image, you want to use the `REDIRECT_STATUS_CODE` and `REDIRECT_CACHE_LIFETIME` env vars.

If you are not using the docker image, then run `bin/set-option` to change those two values.

> **Matomo integration**
> What do you think about a matomo integration?
> In some cases it is necessary to have statistics from a common accepted instance like matomo.

I'm not sure if I follow. Do you mean being able to send Shlink starts to a matomo instance.

(I didn't know matomo, but I have googled it, hehe)

**Reply by tedo-develop** [reply]
Acelaya,

thanks for your reply. 

> You may want to subscribe to #1104

I do and wait for implementation 👍 

> n order to change this, if you are using the docker image, you want to use the REDIRECT_STATUS_CODE and REDIRECT_CACHE_LIFETIME env vars.

I will try it.

> I'm not sure if I follow. Do you mean being able to send Shlink starts to a matomo instance.

Matomo was just an example (we use it). And yes would be nice if we have the option to choose an "common accepted instance" like Google Analytics or Matomo. If Shlink gets an redirect request -> request to 3 Party Application, maybe a good idea :)?

**Reply by acelaya** [reply]
> Matomo was just an example (we use it). And yes would be nice if we have the option to choose an "common accepted instance" like Google Analytics or Matomo. If Shlink gets an redirect request -> request to 3 Party Application, maybe a good idea :)?

This has been discussed in the past, and whether I absolutely agree it would be something nice to have, it requires a ton of work, I'm afraid. I wouldn't expect it any time soon.
